src/frontends/onnx/frontend/src/core/tensor.hpp

```diff
@@ -288,14 +296,16 @@ class Tensor {
                    ONNX_NAMESPACE::TensorProto_DataLocation::TensorProto_DataLocation_EXTERNAL;
     }
 
-    std::string load_external_data() const {
-        const auto tensor_external_data = detail::TensorExternalData(*m_tensor_proto);
-        return tensor_external_data.load_external_data(m_model_dir);
-    }
-
     template <typename T>
     std::vector<T> get_external_data() const {
-        return detail::__get_raw_data<T>(load_external_data(), m_tensor_proto->data_type());
+        const auto ext_data = detail::TensorExternalData(*m_tensor_proto);
+        std::shared_ptr<ngraph::runtime::AlignedBuffer> buffer = nullptr;
+        if (m_enable_mmap) {
+            buffer = ext_data.load_external_mmap_data(m_model_dir);
```

Comment:
why do we open a file/directory for each `Constant`? why can't we prepare streams in advance?
@gkrivor made a great note about this.

Reply:
where can I find `load_external_mmap_data`? In your changes it is absent.
What mechanism provides to have stream opened for mmap while `ov::Model` exists?

Reply:
> why do we open a file/directory for each Constant? why can't we prepare streams in advance?
@gkrivor made a great note about this.

Agree, that it's a good optimization to keep streams open in advance. Are you doing it separately? I can prepare a ticket.
> where can I find load_external_mmap_data? In your changes it is absent.

You mean implementation of this method? If yes - https://github.com/mbencer/openvino/blob/051aa8dd89f23331bd4ca591e6359be41a146a9c/src/frontends/onnx/frontend/src/utils/tensor_external_data.cpp#L37
> What mechanism provides to have stream opened for mmap while ov::Model exists?

The lifetime of such mmap memory is based on `HandleHolder` which exists until ov::Constant shared ptr (created from runtime::SharedBuffer) has references. In `HandleHolder` dtor we call munmap (or equivalent for win) - https://github.com/mbencer/openvino/blob/051aa8dd89f23331bd4ca591e6359be41a146a9c/src/common/util/src/os/lin/lin_mmap_object.cpp#L95C13-L95C19

Reply:
@rkazants, about storing an opened handles. I'm digging into same way and looks like InputModel (let's imagine, with links on an opened mmap handles) are stored in ov::Model->m_shared_object while ov::Model exists (by desing). So, I hope we don't need to store it separately in ov::Model. As I investigated and discussed with author - same approach is used in IR Frontend.